src/evo/deterministicmns.cpp

```diff
@@ -1101,6 +1291,40 @@ static bool CheckService(const ProTx& proTx, CValidationState& state)
     return true;
 }
 
+template <typename ProTx>
+static bool CheckPlatformFields(const ProTx& proTx, CValidationState& state)
+{
+    if (proTx.platformNodeID.IsNull()) {
+        return state.Invalid(ValidationInvalidReason::TX_BAD_SPECIAL, false, REJECT_INVALID, "bad-protx-platform-nodeid");
+    }
+
+    static int mainnetPlatformP2PPort = CreateChainParams(CBaseChainParams::MAIN)->GetDefaultPlatformP2PPort();
+    if (Params().NetworkIDString() == CBaseChainParams::MAIN) {
+        if (proTx.platformP2PPort != mainnetPlatformP2PPort) {
+            return state.Invalid(ValidationInvalidReason::TX_BAD_SPECIAL, false, REJECT_INVALID, "bad-protx-platform-p2p-port");
+        }
+    } else if (proTx.platformP2PPort == mainnetPlatformP2PPort || proTx.platformP2PPort < 1 || proTx.platformP2PPort > std::numeric_limits<uint16_t>::max()) {
```

Comment:
we don't do a numeric limits for the normal p2p port right? This check is not needed as the type is uint16_t, it'll always be true